src/log.rs

```diff
@@ -27,3 +27,26 @@ macro_rules! ngx_log_debug_http {
         $crate::ngx_log_debug!(log, $($arg)*);
     }
 }
+
+/// Log with appropriate debug mask.
+///
+/// When the request logger is available `ngx_log_debug_http` can be used for `NGX_LOG_DEBUG_HTTP` masks.
+/// This macro is useful when other masks are necessary or when the request logger is not
+/// conveniently accessible.
+///
+/// See https://nginx.org/en/docs/dev/development_guide.html#logging for details and available
+/// masks.
+#[macro_export]
+macro_rules! ngx_log_debug_mask {
+    ( $mask:expr, $log:expr, $($arg:tt)* ) => {
+        let log_level = unsafe { (*$log).log_level };
+        if log_level & $mask as usize != 0 {
```

Comment:
Probably, it is a bit hard to understand masks here.


The debug masks are:
```
NGX_LOG_DEBUG_CORE
NGX_LOG_DEBUG_ALLOC
NGX_LOG_DEBUG_MUTEX
NGX_LOG_DEBUG_EVENT
NGX_LOG_DEBUG_HTTP
NGX_LOG_DEBUG_MAIL
NGX_LOG_DEBUG_STREAM
```
and in the generated bindings are:
```rust
pub const NGX_LOG_DEBUG_CORE: u32 = 16;
pub const NGX_LOG_DEBUG_ALLOC: u32 = 32;
pub const NGX_LOG_DEBUG_MUTEX: u32 = 64;
pub const NGX_LOG_DEBUG_EVENT: u32 = 128;
pub const NGX_LOG_DEBUG_HTTP: u32 = 256;
pub const NGX_LOG_DEBUG_MAIL: u32 = 512;
pub const NGX_LOG_DEBUG_STREAM: u32 = 1024;
````
so with that macro it is really easy to make a mistake when using it say mask == 2.
let's think how can we improve this and make it easier/more idiomatic to use it. 

one option is that the mask is an enum...

Reply:
another options is not original. we have extra macro_rules:
```
ngx_log_debug_http
ngx_log_debug_core, 
ngx_log_debug_XYZ
```
and these would use internally ngx_log_debug_mask`.
like `ngx_log_debug_http` uses `ngx_log_debug` already

Reply:
Enums are pretty powerful in Rust, but the logger is still storing log_level as a bit field. So we might need to implement the BitOps traits for them.

It's verbose, but I think the simplest and most foolproof way is a function for each mask. _http, _event, _core, etc. If you're ok with this second approach, I think it'll most easily understood by users.

Reply:
There is a safer enum based version with macro pattern matching in PR #40.